apps/settings/test/unit/modules/bluetooth/bluetooth_context_test.js

```diff
@@ -0,0 +1,1013 @@
+        assert.isTrue(btContext._updateStatus.called);
+        assert.isTrue(btContext._updateStatus.calledWith(mockAdapter.state));
+        assert.isFalse(btContext.startDiscovery.called);
+      });
+    });
+
+    suite('address changed ', function() {
+      setup(function() {
+        evt = {
+          attrs: ['address']
+        };
+      });
+
+      test('address should be called with new address ', function() {
+        btContext._onAdapterAttributeChanged(mockAdapter, evt);
+        assert.equal(btContext.address, mockAdapter.address);
+      });
+    });
+
+    suite('name changed ', function() {
+      setup(function() {
+        evt = {
+          attrs: ['name']
+        };
+      });
+
+      test('name should be called with new name ', function() {
+        btContext._onAdapterAttributeChanged(mockAdapter, evt);
+        assert.equal(btContext.name, mockAdapter.name);
+      });
+    });
+
+    suite('discoverable changed ', function() {
+      setup(function() {
+        evt = {
+          attrs: ['discoverable']
+        };
+      });
+
+      test('discoverable should be called with new state ', function() {
+        btContext._onAdapterAttributeChanged(mockAdapter, evt);
+        assert.equal(btContext.discoverable, mockAdapter.discoverable);
+      });
+    });
+
+    suite('discovering changed ', function() {
+      setup(function() {
+        evt = {
+          attrs: ['discovering']
+        };
+      });
+
+      test('discovering should be called with new state ', function() {
+        btContext._onAdapterAttributeChanged(mockAdapter, evt);
+        assert.equal(btContext.discovering, mockAdapter.discovering);
+      });
+    });
+  });
+
+  suite('_onAdapterDevicepaired > ', function() {
+    var mockAdapter;
+    setup(function() {
+      mockAdapter = {};
+      this.sinon.stub(btContext, '_refreshPairedDevicesInfo');
+    });
+
+    test('_refreshPairedDevicesInfo() should be called with adapter ',
+    function() {
+      btContext._onAdapterDevicepaired(mockAdapter);
+      assert.isTrue(btContext._refreshPairedDevicesInfo.called);
+      assert.isTrue(btContext._refreshPairedDevicesInfo.calledWith(
+        mockAdapter));
+    });
+  });
+
+  suite('_onAdapterDeviceunpaired > ', function() {
+    var mockAdapter;
+    setup(function() {
+      mockAdapter = {};
+      this.sinon.stub(btContext, '_refreshPairedDevicesInfo');
+    });
+
+    test('_refreshPairedDevicesInfo() should be called with adapter ',
+    function() {
+      btContext._onAdapterDeviceunpaired(mockAdapter);
+      assert.isTrue(btContext._refreshPairedDevicesInfo.called);
+      assert.isTrue(btContext._refreshPairedDevicesInfo.calledWith(
+        mockAdapter));
+    });
+  });
+
+  suite('_onDefaultAdapterChanged > ', function() {
+    var newAdapter, oldAdapter;
+    setup(function() {
+      newAdapter = {};
+      oldAdapter = {};
+      this.sinon.stub(btContext, '_defaultAdapter');
```

Comment:
`_defaultAdapter` is a property so we don't use stub on it. Please set it as mock object.